src/expression/comparison_expression.py

```diff
@@ -32,41 +33,43 @@ def __init__(self, exp_type: ExpressionType, left: AbstractExpression,
                          children=children)
 
     def evaluate(self, *args):
-        # evaluate always return batch
-        left_values = self.get_child(0).evaluate(*args).frames
-        right_values = self.get_child(1).evaluate(*args).frames
+        # cast in to numpy array
+        lvalues = self.get_child(0).evaluate(*args).frames.values
+        rvalues = self.get_child(1).evaluate(*args).frames.values
 
-        if len(left_values) != len(right_values):
-            if len(left_values) == 1:
-                left_values = pd.DataFrame(np.repeat(left_values.values,
-                                                     len(right_values),
-                                                     axis=0))
-            elif len(right_values) == 1:
-                right_values = pd.DataFrame(np.repeat(right_values.values,
-                                                      len(left_values),
-                                                      axis=0))
+        if len(lvalues) != len(rvalues):
```

Comment:
What do we want to support here? I am a little lost here.

Reply:
Code here repeats the left or right values if the length is 1. I'm saying we don't need this right? Since now we have the contains/is_contained operators we don't want to do this.

Reply:
Ah,  I got. It is different. Say a predicate UDF(data) = 'car', the UDF(data) will return a dataframe that has `batch_size` numbers of rows, while the 'car' is a single row constant dataframe. So we need to repeat the 'car' dataframe `batch_size` times.

Reply:
Ah, now it makes sense